vtr_flow/scripts/upgrade_arch.py

```diff
@@ -154,7 +154,12 @@ def main():
         if args.debug:
             root.write(sys.stdout, pretty_print=args.pretty)
         else:
-            with open(args.xml_file, "w") as f:
+            # Added the "b" flag to silence the str/bytes
+            # error. Output of script is identical except for some XML
+            # fields are in a different order. This seems to be the
+            # doing of the lxml library and won't be a simple fix if
+            # it causes problems.
```

Comment:
Similarly, stating that the 'b' flag is needed for str/byte error is a good comment.

That the output is identical modulo lxml ordering is something that should go in the commit message. (Someone reading the comment years from now won't know what the reference point for 'identical' is, so best to leave that in the commit messages which give more temporal context).